app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Any
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        cached_key = (func, args)
```

Comment:
Why do you need store func?

Reply:
there is no "store func". If it's about @wraps(func), I've inserted as requested by Dimosphen1
"Apply a decorator from functools to the inner to preserve the wrapped function name and docstring."

Reply:
@Roman-Sokolov-V @ArsenPidhoretskyi asked about the need to store `func` inside if `cached_key` rather than about using `wraps` decorator. Since storing function inside of the cache is not obligatory in this task, it was a point of this question. However, we had a discussion during a Q&A session where we decided to store `func` explicitly to prevent cases when two different functions may be decorated using `cache` and have the same arguments but different logic so that they would not contradict one another.